### DISCUSSION TITLE
Is there a way to implement an infinit scroller without adding javascript ?

### DISCUSSION THREAD

**Discussion by MayasMess** [django-commons/django-unicorn#270] [Q&A] [answered]
Hello hello, everything is in the title :).
I'm creating a todo list with all the crud fonctionnallities and it's working really well. The only thing I didn't find how to do without adding javascript, is the infinit scroller.

![image](https://user-images.githubusercontent.com/51958712/132400166-bfc2942b-da46-4e4a-ab4d-e87886f74e66.png)

Cheers!

**Comment by adamghill** [comment]
I can't think of a way to do this in `Unicorn` currently without _some_ custom JavaScript or without a button or link that gets clicked. But, you could create an [IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API) which would be relatively straight-forward and then it could use https://www.django-unicorn.com/docs/actions/#calling-methods to call a view method.

However, would something like the following be useful if it was added to `Unicorn`?

```html
{% for object in objects %}
<div>{{ object }}</div>
{% endfor %}

<template unicorn:visible="get_new_objects" />
```

**Reply by MayasMess** [reply]
Hi Adam, thank you for your answer.

Will the template tag get trigerred when it is visible on the page and then call the method get_new_objects automatically ?
If it's the case, yeah it would be great.

**Reply by adamghill** [reply]
>Will the template tag get trigerred when it is visible on the page and then call the method get_new_objects automatically

Yep! That would be the idea. I'll see how easy it is to add that functionality.

**Reply by adamghill** [reply]
https://github.com/adamghill/django-unicorn/pull/271 has a proof of concept and it seems to work pretty well, but I need to do some more testing.

**Comment by adamghill** [comment, marked as answer]
This is now released in 0.35.0. Documentation is located here: https://www.django-unicorn.com/docs/visibility/. Let me know if you run into any issues with it!

**Reply by MayasMess** [reply]
Thank you Adam, I'll try it today 👍

**Reply by MayasMess** [reply]
So I added the span tag after the for loop in the template like in the documentation:
![image](https://user-images.githubusercontent.com/51958712/132851872-896ff716-ad47-4ece-8b55-993d7d794065.png)

But the function is not trigerred when I scroll down to it and make it visible.

This is the view:
![image](https://user-images.githubusercontent.com/51958712/132852079-30e12bbe-a31e-4af1-9c57-6af7129b20ce.png)

Is it supposed to work this way ?

**Comment by adamghill** [comment]
That looks like it should work. Have you updated the version of Unicorn installed to 0.35.0? Do you see any web dev console or Python errors?
________________________________

**Reply by MayasMess** [reply]
I tried it out on another laptop and it's working well. maybe it was the version that didn't update correctly :/ even though I checked my virtual env and the changes you made were there.
Anyway ^^ thanks a lot :)

**Comment by algameko** [comment]
It work very well.. but why the state loading is not triggered?
